server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java

```diff
@@ -47,30 +47,30 @@
 public class QProfileFactory {
 
   private final DbClient db;
-  private final DefaultOrganizationProvider defaultOrganizationProvider;
 
-  public QProfileFactory(DbClient db, DefaultOrganizationProvider defaultOrganizationProvider) {
+  public QProfileFactory(DbClient db) {
     this.db = db;
-    this.defaultOrganizationProvider = defaultOrganizationProvider;
   }
 
   // ------------- CREATION
 
-  QualityProfileDto getOrCreate(DbSession dbSession, QProfileName name) {
+  QualityProfileDto getOrCreate(DbSession dbSession, String organizationUuid, QProfileName name) {
```

Comment:
couldn't this take an `OrganisationDto` rather than a `String`?
strong type and put constraint on calling code to retrieve a `OrganizationDto` from DB (and check that the organization exists)